x/mongo/driver/operation.go

```diff
@@ -755,7 +762,7 @@ func (op Operation) Execute(ctx context.Context) error {
 		// server or connection to nil to continue using the same connection.
 		if batching && len(op.Batches.Documents) > 0 {
 			if retrySupported && op.Client != nil && op.RetryMode != nil {
-				if *op.RetryMode > RetryNone {
+				if *op.RetryMode != RetryNone && *op.RetryMode != RetryTimeout {
 					op.Client.IncrementTxnNumber()
 				}
```

Comment:
I was confused about why we only increment the transaction number when retries are enabled until I read the commnet [here](https://github.com/mongodb/mongo-go-driver/blob/1e144a8ea68ece086da6a1658238a6a05c0d5ed2/x/mongo/driver/operation.go#L469-L474), specifically this section:
> Calling IncrementTxnNumber() for server descriptions or topologies that do not support retries (e.g. standalone topologies) will cause server errors. Only do this check for the first attempt to keep retried writes in the same transaction.

Optional: Consider including that information in a comment in this section as well to reduce confusion.

Reply:
Ah thank you for looking into it! Added a similar comment above.